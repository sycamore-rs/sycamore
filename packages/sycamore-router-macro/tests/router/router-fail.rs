use sycamore_router::Route;

#[derive(Route)]
struct Routes1 {}

// Missing #[not_found]
#[derive(Route)]
enum Routes2 {}

#[derive(Route)]
enum Routes3 {
    #[not_found]
    NotFound(i32), // Cannot have field
}

#[derive(Route)]
enum Routes4 {
    #[to("<capture>")]
    Path, // Missing capture field
    #[not_found]
    NotFound,
}

#[derive(Route)]
enum Routes5 {
    #[to("<capture>")]
    Path {}, // Missing capture field
    #[not_found]
    NotFound,
}

#[derive(Route)]
enum Routes6 {
    #[to("<capture>")]
    Path { not_capture: u32 }, // Wrong capture field name
    #[not_found]
    NotFound,
}

#[derive(Route)]
enum Routes7 {
    #[to("<a>/<b>")]
    Path { b: u32, a: u32 }, // Wrong order
    #[not_found]
    NotFound,
}

#[derive(Route)]
enum Routes8 {
    #[to("<a/b>")] // `a/b` is not an identifier
    Path { a: u32 },
    #[not_found]
    NotFound,
}

fn main() {}
