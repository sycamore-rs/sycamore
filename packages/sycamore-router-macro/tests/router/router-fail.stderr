error: Route can only be derived on enums
 --> tests/router/router-fail.rs:4:1
  |
4 | struct Routes1 {}
  | ^^^^^^

error: not found route not specified
 --> tests/router/router-fail.rs:8:1
  |
8 | enum Routes2 {}
  | ^^^^

error: not found route cannot have any fields
  --> tests/router/router-fail.rs:13:13
   |
13 |     NotFound(i32), // Cannot have field
   |             ^^^^^

error: mismatch between number of capture fields and variant fields (found 1 capture field(s) and 0 variant field(s))
  --> tests/router/router-fail.rs:16:10
   |
16 | #[derive(Route)]
   |          ^^^^^
   |
   = note: this error originates in the derive macro `Route` (in Nightly builds, run with -Z macro-backtrace for more info)

error: mismatch between number of capture fields and variant fields (found 1 capture field(s) and 0 variant field(s))
  --> tests/router/router-fail.rs:27:10
   |
27 |     Path {}, // Missing capture field
   |          ^^

error: capture field name mismatch (expected `capture`, found `not_capture`)
  --> tests/router/router-fail.rs:35:12
   |
35 |     Path { not_capture: u32 }, // Wrong capture field name
   |            ^^^^^^^^^^^

error: capture field name mismatch (expected `a`, found `b`)
  --> tests/router/router-fail.rs:43:12
   |
43 |     Path { b: u32, a: u32 }, // Wrong order
   |            ^

error: mismatch between number of capture fields and variant fields (found 0 capture field(s) and 1 variant field(s))
  --> tests/router/router-fail.rs:51:10
   |
51 |     Path { a: u32 },
   |          ^^^^^^^^^^
